optics-core/src/Optics/Fold.hs

```diff
@@ -675,5 +678,29 @@ lookupOf :: (Is k A_Fold, Eq a) => Optic' k is s (a, v) -> a -> s -> Maybe v
 lookupOf o a = foldrOf o (\(a', v) next -> if a == a' then Just v else next) Nothing
 {-# INLINE lookupOf #-}
 
+-- | Given a 'Fold' that knows how to locate immediate children, retrieve all of
+-- the transitive descendants of a node, including itself.
+universeOf :: Is k A_Fold => Optic' k is a a -> a -> [a]
+universeOf l = go
+  where
+    go a = a : foldMapOf l go a
```

Comment:
this is slow for trees. `++` can easily make this quadratic. I don't know why DList = Endo [a] is not used here instead.

Reply:
Can you give an example?

> I don't know why DList = Endo [a] is not used here (and in lens) instead.

Me neither, I assume since `lens` has it like that it's fine :thinking: It can be optimized later if necessary.

Reply:
See https://github.com/ekmett/lens/pull/994

Reply:
I updated PR to reflect changes made to lens.